### PR TITLE
feat(connector): added template code for datatrans

### PR DESCRIPTION
**Pull request by awasthi21**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description

Added Template code for Datatrans. Ran add_connector.sh script

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [x] This PR modifies application configuration/environment variables

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

File: crates/router/src/core/payments/flows.rs [hunk 1]
```diff
@@ -166,6 +166,7 @@ default_imp_for_complete_authorize!(
     connector::Coinbase,
     connector::Cryptopay,
     connector::Dlocal,
+    connector::Datatrans,
```
Comment (SamraatBansal) [inline crates/router/src/core/payments/flows.rs:169]: Should be in alphabetical order in all the places

**Review by SamraatBansal — CHANGES_REQUESTED** [review]

File: crates/api_models/src/enums.rs [hunk 2]
```diff
@@ -89,6 +89,7 @@ pub enum Connector {
     Cryptopay,
     Cybersource,
     Dlocal,
+    // Datatrans,
```
[the remark this answers is not in the document]
Comment (awasthi21) [reply crates/api_models/src/enums.rs:92]: Made the required changes

File: crates/router/src/core/payments/flows.rs (hunk 1 above)
Comment (awasthi21) [reply crates/router/src/core/payments/flows.rs:169]: fixed the required changes

**Review by AkshayaFoiger — APPROVED** [review]

**Review by SamraatBansal — APPROVED** [review]

File: crates/router/tests/connectors/sample_auth.toml [hunk 3]
```diff
@@ -232,4 +232,6 @@ api_key="API Key"
 api_key="API Key"
 
 [adyenplatform]
+
```
Comment (sai-harsha-vardhan) [inline crates/router/tests/connectors/sample_auth.toml:235]: api_key is migging for adyenplatform here

Comment (awasthi21) [reply crates/router/tests/connectors/sample_auth.toml:236]: Fixed that

**Review by SamraatBansal — APPROVED** [review]

**Review by SanchithHegde — APPROVED** [review]

**Pull request merged**